server/sockets.ts

```diff
@@ -21,6 +21,39 @@ import {IPTools} from './ip-tools';
 type StreamWorker = ProcessManager.StreamWorker;
 type ChannelID = 0 | 1 | 2 | 3 | 4;
 
+export function extractChannelMessages(message: string) {
+	const channelMessages = new Map<ChannelID | -1, string[]>(
```

Comment:
I'm not personally sold on returning a map. This function could:
- accept a list of ChannelIDs and then return the logs in the order they were provided
- accept nothing but return a list of logs in ascending order (-1, 0, 1, 2, 3, 4)

Not sure of the best approach here.